storage_v2/spanstore/reader.go

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package spanstore
+
+import (
+	"context"
+	"time"
+
+	"github.com/jaegertracing/jaeger/model"
+)
+
+// Reader finds and loads traces and other data from storage.
+type Reader interface {
+	// GetTrace retrieves the trace with a given id.
+	//
+	// If no spans are stored for this trace, it returns ErrTraceNotFound.
+	GetTrace(ctx context.Context, traceID model.TraceID) (*model.Trace, error)
+
+	// GetServices returns all service names known to the backend from spans
+	// within its retention period.
+	GetServices(ctx context.Context) ([]string, error)
+
+	// GetOperations returns all operation names for a given service
+	// known to the backend from spans within its retention period.
+	GetOperations(ctx context.Context, query OperationQueryParameters) ([]Operation, error)
```

Comment:
```suggestion
	GetOperations(ctx context.Context, query GetOperationsParameters) ([]Operation, error)
```
this makes it more consistent with the Protobuf API